### ISSUE TITLE
Parsing single array param feature request

### ISSUE THREAD

**Issue by AuthorProxy** [ljharb/qs#305]
I have some location which accepts multiple `orderItemIds`, in case of single param value `qs.parse("...?orderItemIds=1119012", { ignoreQueryPrefix: true })`, I receive string instead of an array. How can I specify that I want array instead of string, to prevent additional code checks?

**Comment by ljharb** [comment]
Can you provide more information? In the example you gave, there's only one `orderItemIds`, so it's not clear on what you want vs what qs already does.

**Comment by AuthorProxy** [comment]
I load a page which can contain multiple orderItemIds at url:
```
// pageUrl = "...?orderItemIds=1119012&orderItemIds=1119013&orderItemIds=1119014"
let someArray = qs.parse(pageUrl);
// working with array
```

And next I make some work with this data, but when url contains only single orderItemIds, I receive string instead of an array, so I should write something like this:
```

// pageUrl = "...?orderItemIds=1119012"
let parsedArgs = qs.parse(pageUrl);
let someArray = Object.prototype.toString.call(parsedArgs) === '[object String]' ? [parsedArgs] : parsedArgs;
// working with array
```

I would like to see some option like `arrayFormat: 'repeat'`, that let me specify url format and avoid manual type checking.

**Comment by ljharb** [comment]
Use `[]` at the end of the key name (as is common convention) and it will always be an array.

**Comment by AuthorProxy** [comment]
Sure I can, but I get this url by serializing `qs.serialize(arrayFormat: 'repeat')` everywhere, because my backend api don't understand `[]` syntax.

And I want to parse it the same way. It would be conventional with your own syntax (`arrayFormat: 'repeat'`), isn't it?

**Comment by ljharb** [comment]
Yes; also the spec requires that that syntax be supported (but does not require that it be used, when there’s multiple keys, so the string you’re sending is valid without the brackets).

I think that to be able to know a non-bracketed key was an array, we’d have to add some kind of option where you list key names for which this should apply, only for parse. That seems like a lot of complexity to avoid adding the conventional and standard brackets to hit key name (altho i realize that’s the point of the “repeat” format) :-/
